dcos/marathon.py

```diff
@@ -130,6 +130,8 @@ class Client(object):
     """
 
     def __init__(self, marathon_url, timeout=http.DEFAULT_TIMEOUT):
+        if not marathon_url.endswith('/'):
+            marathon_url += '/'
         self._base_url = marathon_url
```

Comment:
I think that os.path.join will give you different results depending on platform ('/' on Linux vs '\' on Windows machines. 

This patch is meant only as quick fix, but if you want to fix this issue globally, instead of _get_marathon_url I would try to patch `_create_url` and substitue
```
return urllib.parse.urljoin(self._base_url, path)
```
with i.e. something similar to what @jgehrcke suggested:
```
'/'.join(c.strip('/') for c in components)
```
I was afraid that it can break other things so I provided a "localized" patch that poses very low risk of breaking stuff.